File: 单例/单例/ViewController.m
//
//  ViewController.m
//  单例
//
//  Created by Houge on 2020/1/12.
//  Copyright © 2020 ZHUNJIEE. All rights reserved.
//

#import "ViewController.h"
#import "Singleton.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    Singleton *single1 = [Singleton shareInstance];
    Singleton *single2 = [Singleton shareInstance];
    Singleton *single3 = [[Singleton alloc] init];
    Singleton *single4 = [Singleton new];
    
    NSLog(@"第一个对象的地址%p",single1);
    NSLog(@"第二个对象的地址%p",single2);
    NSLog(@"第三个对象的地址%p",single3);
    NSLog(@"第四个对象的地址%p",single4);
}


@end

File: AVPlayerDemo/AVPlayerDemo/ViewController.swift
//
//  ViewController.swift
//  AVPlayerDemo
//
//  Created by 侯歌 on 2020/3/31.
//  Copyright © 2020 HouGe. All rights reserved.
//

import UIKit
import AVFoundation

class ViewController: UIViewController {
    lazy var player: AVAudioPlayer = {
        let path = Bundle.main.path(forResource: "test", ofType: "mp3")
        let url = NSURL(fileURLWithPath: path ?? "")
        let player = try! AVAudioPlayer(contentsOf: url as URL)
        player.numberOfLoops = 0
        player.volume = 1.0
        player.prepareToPlay()
        return player
    }()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        let tools = BWPlayerToolsView(frame: CGRect(x: 0, y: 30, width: UIScreen.main.bounds.width, height: 30))
        view.addSubview(tools)
    }

    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
        
        self.player.play()
    }
}

